ci/do_ci.sh

```diff
@@ -341,17 +341,24 @@ elif [[ "$1" == "cmake.legacy.test" ]]; then
 elif [[ "$1" == "cmake.clang_tidy.test" ]]; then
   cd "${BUILD_DIR}"
   rm -rf *
-  export BUILD_ROOT="${BUILD_DIR}"
+  clang-tidy --version
+  LOG_FILE="${BUILD_DIR}/opentelemetry-cpp-clang-tidy.log"
   cmake -S ${SRC_DIR} \
     -B ${BUILD_DIR} \
     -C ${SRC_DIR}/test_common/cmake/all-options-abiv2-preview.cmake  \
     "${CMAKE_OPTIONS[@]}" \
     -DWITH_OPENTRACING=OFF \
     -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations" \
     -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
-    -DCMAKE_CXX_CLANG_TIDY="clang-tidy;--quiet;-p;${BUILD_DIR}"
-  make -j $(nproc)
+    -DCMAKE_CXX_CLANG_TIDY="clang-tidy;--header-filter=.*/opentelemetry-cpp/.*;--exclude-header-filter=.*(internal/absl|third_party|third-party)/.*;--quiet"
```

Comment:
Is there a way to put this logic (header-filter) in the top level file .clang-tidy instead ?

It is desirable to have all the clang-tidy configuration in the same place, independent of the workflow & makefiles.